packages/x-date-pickers-pro/src/internal/utils/valueManagers.ts

```diff
@@ -63,55 +63,48 @@ export const rangeFieldValueManager: FieldValueManager<
         ? { startDate: null, endDate: null }
         : splitDateRangeSections(prevSections);
 
-    const getSections = (newDate: any | null, prevDateSections: RangeFieldSection[] | null) => {
+    const getSections = (
+      newDate: any | null,
+      prevDateSections: RangeFieldSection[] | null,
+      position: RangePosition,
+    ) => {
       const shouldReUsePrevDateSections = !utils.isValid(newDate) && !!prevDateSections;
 
       if (shouldReUsePrevDateSections) {
         return prevDateSections;
       }
 
-      return splitFormatIntoSections(utils, localeText, format, newDate);
-    };
-
-    const rawSectionsOfStartDate = getSections(start, prevDateRangeSections.startDate);
-    const rawSectionsOfEndDate = getSections(end, prevDateRangeSections.endDate);
-
-    const sectionsOfStartDate: RangeFieldSectionWithoutPosition[] = rawSectionsOfStartDate.map(
-      (section, sectionIndex) => {
-        if (sectionIndex === rawSectionsOfStartDate.length - 1) {
+      const sections = splitFormatIntoSections(utils, localeText, format, newDate);
+      return sections.map((section, sectionIndex) => {
+        if (sectionIndex === sections.length - 1) {
```

Comment:
If we do not have `position === 'start'` it woudl have the same behavior as the default `return`. So it's easier to read when all conditions are in the same `if`
```suggestion
        if (sectionIndex === sections.length - 1 && position === 'start') {
```